library/Fission/Internal/MonadDB/Class.hs

```diff
@@ -2,7 +2,5 @@ module Fission.Internal.MonadDB.Class (MonadDB (..)) where
 
 import RIO
 
-import Fission.Internal.MonadDB.Types
-
-class MonadIO m => MonadDB m where
-  runDB :: Transaction m a -> m a
+class MonadIO m => MonadDB transaction m | m -> transaction where
+  runDB :: transaction a -> m a
```

Comment:
What the heck is that `| m -> transaction`? It says that there's only one implementation of this type class for `m`. So `Fission` will _always_ point at `Transaction`, and `TestMock` will _always_ point at `TestTransaction` (or whatever)